rust/src/sip/sip.rs

```diff
@@ -386,6 +567,24 @@ pub unsafe extern "C" fn rs_sip_register_parser() {
             let _ = AppLayerRegisterParser(&parser, alproto);
         }
     } else {
-        SCLogDebug!("Protocol detecter and parser disabled for SIP/UDP.");
+        SCLogDebug!("Protocol detection and parsing disabled for UDP SIP.");
+    }
+
+    // register TCP parser
+    parser.ipproto = core::IPPROTO_TCP;
```

Comment:
I think we should use pattern based detection too, esp since this is a http like proto. We can use `AppLayerProtoDetectPMRegisterPatternCSwPP` to register patterns with a probing parser to validate it. See smb.rs.

Reply:
Giuseppe already answered me in a previous PR that this was tracked as a separate redmine ticket https://redmine.openinfosecfoundation.org/issues/5047